### PR TITLE
feat: Greenhouse app

### PR DESCRIPTION
**Pull request by aar2dee2**
## What does this PR do?
Implements #13148
[Greenhouse](https://www.greenhouse.com/) integration to sync bookings
Currently a draft PR while I wait for access to a Greenhouse sandbox to test this.

Fixes # (issue)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [-] New feature (non-breaking change which adds functionality)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [-] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/app-store/greenhouse/config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "type": "greenhouse_other",
+  "logo": "icon.svg",
+  "url": "https://example.com/link",
+  "variant": "other",
+  "categories": ["other"],
+  "publisher": "aar2dee2",
+  "email": "support@cal.com",
+  "description": "Applicant tracking software",
+  "isTemplate": false,
+  "__createdUsingCli": true,
```
Comment (PeerRich) [inline packages/app-store/greenhouse/config.json:14]:
```suggestion
  "/*": "Don't modify slug - If required, do it using cli edit command",
  "name": "Greenhouse",
  "slug": "greenhouse",
  "type": "greenhouse_other",
  "logo": "icon.svg",
  "url": "https://greenhouse.io",
  "variant": "other",
  "categories": ["other"],
  "publisher": "aar2dee2",
  "email": "support@cal.com",
  "description": "Applicant tracking software",
  "isTemplate": false,
  "__createdUsingCli": true,
```

**Review by PeerRich — DISMISSED** [review]
got a sandbox key now. lets merge this and try it on production, since its behind a feature flag anyway

**Comment by aar2dee2** [comment]
Hi,
Most of the code is in place, have a few Qs.

1. For conferencing apps, the app gets added as a `location` for the event. For calendar apps, the app can be chosen as the calendar for a given event. Greenhouse falls in neither category but works similarly, you can create/edit/delete interviews - how will this work with a Cal event? (The Basecamp integration looks similar but I don't know how it operates)
2. I added another parameter to the app keys schema in `zod.ts` in the greenhouse directory, but this isn't showing up in app settings in admin settings. Do I need to run a command to update this?
3. Do we need to create webhooks in Greenhouse? There is only the [Interview Deleted event](https://developers.greenhouse.io/webhooks.html#interview-deleted) related to scheduled interviews for which Greenhouse will send notifications.

**Comment by joeauyeung** [comment]
> Hi, Most of the code is in place, have a few Qs.
> 
> 1. For conferencing apps, the app gets added as a `location` for the event. For calendar apps, the app can be chosen as the calendar for a given event. Greenhouse falls in neither category but works similarly, you can create/edit/delete interviews - how will this work with a Cal event? (The Basecamp integration looks similar but I don't know how it operates)
> 2. I added another parameter to the app keys schema in `zod.ts` in the greenhouse directory, but this isn't showing up in app settings in admin settings. Do I need to run a command to update this?
> 3. Do we need to create webhooks in Greenhouse? There is only the [Interview Deleted event](https://developers.greenhouse.io/webhooks.html#interview-deleted) related to scheduled interviews for which Greenhouse will send notifications.

1. I think this can live under locations. Zoom also has a dashboard that displays all future calls.
2. Running `yarn app-store:build` should work.
3. How would the webhooks for Greenhouse work? Would it be up to us as the connected app to send the deleted booking event to Greenhouse to their "Deleted Interview" webhook? Is there any additional work we would have to do?

**Comment by joeauyeung** [comment]
> @joeauyeung why would it be under location? Greenhouse does not come with its own video client, does it?

Greenhouse also has the same video integrations. We should have a single source of truth for video links. If we choose to have it on Cal we would need clear messaging to disable the video link generating on Greenhouse.

**Comment by keithwillcode** [comment]
Going to postpone this to 4.0 as we wait to have an implementation of this not requiring users getting API keys but instead be able to use OAuth.

**Pull request closed without merging**